### ISSUE TITLE
Use Fast.api with fastai

### ISSUE THREAD

**Issue by khs123** [tiangolo/fastapi#2213]
### First check

*
I want to use fast.api as the api for my model predict, my model uses fast.ai technology but I have trouble combining the two above, I will describe the code below.

```Python
import asyncio
from typing import Optional

from PIL import Image
from fastai.learner import load_learner
from fastapi import FastAPI

from face_analyst import faceAnalysis

app = FastAPI()
siamese = load_learner('models/ir152_balance-1.pkl', cpu=True)


@app.get("/")
def read_root():
    faceAnalysis.init_app(siamese)

    s_img = Image.open('mint.jpg').convert('RGB')
    sfaces, sboxes, _ = faceAnalysis.evlo_face_detector(s_img)

    return {"Hello": "World"}
```

### Description
WHEN I RUN SERVER With uvicorn, there will be an error related to load_learner related to the main process
```
File "./fast_api.py", line 7, in <module>
    siamese = load_learner('models/ir152_balance-1.pkl', cpu=True)
  File "/home/eway2020/.pyenv/versions/3.8.0/envs/face_analyst/lib/python3.8/site-packages/fastai/learner.py", line 549, in load_learner
    res = torch.load(fname, map_location='cpu' if cpu else None)
  File "/home/eway2020/.pyenv/versions/3.8.0/envs/face_analyst/lib/python3.8/site-packages/torch/serialization.py", line 584, in load
    return _load(opened_zipfile, map_location, pickle_module, **pickle_load_args)
  File "/home/eway2020/.pyenv/versions/3.8.0/envs/face_analyst/lib/python3.8/site-packages/torch/serialization.py", line 842, in _load
    result = unpickler.load()
AttributeError: Can't get attribute 'siamese_splitter' on <module '__mp_main__' from '/home/eway2020/.pyenv/versions/face_analyst/bin/uvicorn'>

```
### Environment

* OS: Linux
* FastAPI Version 0.61.1
* Python version 3.6.8

**Comment by includeamin** [comment]
I think the version of your python used is 3.8, not 3.6.8

**Comment by eseglem** [comment]
This is an issue with how you are trying to load your learner. You don't have a `siamese_splitter` function for it to load: https://docs.fast.ai/learner#load_learner

**Comment by ycd** [comment]
@includeamin does Python version makes any difference in his situation?

**Comment by includeamin** [comment]
> @includeamin does Python version makes any difference in his situation?

Not exactly.
I Just try to fix wrong information in issue.:)

**Comment by khs123** [comment]
> > @includeamin does Python version makes any difference in his situation?
> 
> Not exactly.
> I Just try to fix the wrong information in the issue.:)

Yes ! i am wrong but all versions are the same. Back to problem, Do you have any ideas to fix this error ?

**Comment by includeamin** [comment]
> > > @includeamin does Python version makes any difference in his situation?
> > 
> > 
> > Not exactly.
> > I Just try to fix the wrong information in the issue.:)
> 
> Yes ! i am wrong but all versions are the same. Back to problem, Do you have any ideas to fix this error ?

maybe I can help you when I have the source code. `from face_analyst import faceAnalysis` and `models/ir152_balance-1.pkl`.
Here is my email: aminjamal10@gmail.com. You can email them . `if you want`

**Comment by muellerzr** [comment]
@ycd I've worked all day on this and I've narrowed down the issue:

So here's what we have going on, we're trying to import in our app while a multiprocessing pool is active. As a result not every pool winds up having the function we want, leading to our errors. What I can't (for the life of me) figure out is how to have our `learn` object be created and returned while in the middle of the processes. This should be impossible I _think_. 

Let me know if that helps a little

**Comment by hari-az** [comment]
Hi All, 

I found a workaround which will work in this case, The problem I understood is that FastAI will looking for the support functions(siamese_splitter) in the __main__ namespace to load the pickle. to fix this you have to import the current running namespace. In python you can do it by simply importing __main__ like import __main__ and then add the support functions (siamese_splitter) to the __main__ namespace like __main__. siamese_splitter = siamese_splitter. Now it will work.

code changed:

import asyncio
from typing import Optional
import __main__

from PIL import Image
from fastai.learner import load_learner
from fastapi import FastAPI

from face_analyst import faceAnalysis

app = FastAPI()
#your support function for the loading the pickle
def siamese_splitter():
     pass
__main__.siamese_splitter = siamese_splitter
siamese = load_learner('models/ir152_balance-1.pkl', cpu=True)


@app.get("/")
def read_root():
    faceAnalysis.init_app(siamese)`

    s_img = Image.open('mint.jpg').convert('RGB')
    sfaces, sboxes, _ = faceAnalysis.evlo_face_detector(s_img)

    return {"Hello": "World"}

without the support functions you used for the training or while exporting the model as pickle you cannot load the pickle.

hope this helps you.
